### DISCUSSION TITLE
Nonlinear projection (e.g., fisheye camera)

### DISCUSSION THREAD

**Discussion by jwwalker** [google/filament#3617] [Q&A]
In the OpenGL world, I've done some experimentation with fisheye and all-seeing cameras, which use a nonlinear projection.  That doesn't play very well with the linear interpolation between the vertex shader and the fragment shader.  So I've used a geometry shader to subdivide triangles that were too big in screen space.  How would you handle something like that in Filament?  I suppose one alternative would be to do offscreen renders to create a cube map texture on the fly, and then transform that in the fragment shader.  But it seems like rendering everything 6 times would inevitably be a performance problem.

**Comment by romainguy** [comment]
More importantly this kind of projection might not be compatible with our lighting system.

**Reply by romainguy** [reply]
@pixelflinger
